addons/default-costume-editor-color/addon.json

```diff
@@ -0,0 +1,48 @@
+{
+  "name": "Customize default costume editor colors",
+  "description": "Changes the default colors and outline size used by the costume editor.",
+  "credits": [
+    {
+      "name": "GarboMuffin"
+    }
+  ],
+  "userscripts": [
+    {
+      "url": "userscript.js",
+      "matches": ["projects"]
+    }
+  ],
+  "settings": [
+    {
+      "name": "Default fill color",
+      "id": "fill",
+      "type": "color",
+      "default": "#9966FF"
+    },
+    {
+      "name": "Default outline color",
+      "id": "stroke",
+      "type": "color",
+      "default": "#000000"
+    },
+    {
+      "name": "Default outline size",
+      "id": "strokeSize",
+      "type": "integer",
+      "min": 0,
+      "max": 100,
+      "default": 4
+    },
+    {
+      "name": "Use previous color instead of default color when resetting color after switching tools",
+      "id": "persistence",
```

Comment:
The persistence only applies to the fill color, not to outline and stroke, correct?

Reply:
It's currently supposed to apply to fill, stroke color, and stroke width.

Reply:
I think I'm confused since I don't know under what situations it's supposed to return to the previous colors and width. How are you testing this?

Reply:
For fill:

1. Make a new project and open the costume editor on the cat
2. Change the default fill color to something that isn't the default
3. Do Ctrl+A
4. Notice how Scratch has reset your fill color to a generic "mixed color" icon
5. Switch to the paint brush tool. Scratch would normally reset your color to the default purple, but with the option enabled, it will instead use the color you had selected before step 3.

For stroke/outline: Same as above but using vector rectangle tool instead of paint brush tool.

It's buggy right now, but that's what it's supposed to do.